citesphere/src/main/java/edu/asu/diging/citesphere/core/service/impl/CitationManager.java

```diff
@@ -379,4 +380,15 @@ public CitationPage getPrevAndNextCitation(IUser user, String groupId, String co
         }
         return result;
     }
-}
+
+    @Override
+    public void deleteCitations(IUser user, String groupId, List<String> citationList) throws ZoteroConnectionException, ZoteroHttpStatusException {
+        for(String element : citationList) {
+            Optional<ICitation> citation = citationStore.findById(element);
+            if(citation.isPresent()) {
+                citationStore.delete(citation.get());
+                zoteroManager.deleteCitation(user, groupId, citation.get());
```

Comment:
let's make one deletion request instead of several. Zotero API allows deleting up to 50 items with one request:
https://www.zotero.org/support/dev/web_api/v3/write_requests#deleting_multiple_items